lib/DXIL/DxilUtil.cpp

```diff
@@ -1215,6 +1217,32 @@ Value *TryReplaceBaseCastWithGep(Value *V) {
   return nullptr;
 }
 
+// returns true if the function call is an intermediate or DXIL op 
+// that has no side effects, for functions not marked ReadNone or ReadOnly.
+bool DxilOpFunctionHasNoSideEffects(Instruction *I) {
+  if (CallInst *CI = dyn_cast<CallInst>(I)) {
+    if (CI->onlyReadsMemory()) return false;
+
+    bool isDxilOp = hlsl::OP::IsDxilOpFunc(CI->getCalledFunction());
+    if (!isDxilOp) return false;
```

Comment:
Another nit. Given that the return value is only used this once, it would seem more natural to include the call in the conditional and leave out the variable entirely.